File: app/src/main/java/com/example/apkversionchecker/data/AppVersion.java
package com.example.apkversionchecker.data;

import android.text.TextUtils;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.dhy.versionchecker.IVersion;
import com.example.apkversionchecker.BuildConfig;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class AppVersion implements Serializable, IVersion {
    @SerializedName("packagename")
    public String packageName;
    @SerializedName("versioncode")
    private int versionCode;

    @SerializedName("versiontype")
    private String type;
    @SerializedName("versionname")
    private String versionName;
    @SerializedName("forceupdate")
    private int forceUpdate;

    @Override
    public boolean isForceUpdate() {
        return forceUpdate == 1;
    }

    @Override
    public long getSize() {
        return size;
    }

    @NonNull
    @Override
    public String getUrl() {
        return url;
    }

    public String patchUrl;

    @Nullable
    @Override
    public String getPatchUrl() {
        return patchUrl;
    }

    @Override
    public void setPatchUrl(@Nullable String url) {
        patchUrl = url;
    }

    public long patchSize;

    @Override
    public long getPatchSize() {
        return patchSize;
    }

    @Override
    public void setPatchSize(@NonNull Long size) {
        patchSize = size;
    }

    @NonNull
    public String getVersionName() {
        if (versionName != null) {
            return versionName;
        } else return "";
    }

    @Override
    public int getNewVersionCode() {
        return versionCode;
    }

    @Override
    public int getOldVersionCode() {
        return BuildConfig.VERSION_CODE;
    }

    @Override
    public String getOldVersionName() {
        return BuildConfig.VERSION_NAME;
    }

    @Override
    public boolean isNew() {
        return BuildConfig.VERSION_CODE < versionCode;
    }

    @Override
    public String getLog() {
        return TextUtils.isEmpty(message) ? "" : message;
    }

    private String url;
    /**
     * 软件大小，单位 byte
     */
    @SerializedName("packagesize")
    public long size;
    private String message;
}

File: app/src/main/java/com/example/apkversionchecker/AppVersion.kt
package com.example.apkversionchecker

import com.dhy.versionchecker.IVersion

class AppVersion : IVersion {
    companion object {
        var forceUpdate = false
    }

    override fun getPatchSize(): Long {
        return size
    }

    override fun getPatchUrl(): String? {
        return null
    }

    override fun setPatchSize(size: Long) {

    }


    override fun getVersionCode() = 0
    override fun setPatchUrl(url: String?) {

    }

    override fun getSize() = 4882533L

    override fun getUrl(): String {
        return "http://apk.wwvas.com:10000/apk/com.example.apkversionchecker-v1.0_c20_t0.apk"
    }

    override fun getCurrentVersionCode(): Int {
        return BuildConfig.VERSION_CODE
    }

    override fun isForceUpdate() = forceUpdate

    override fun getVersionName() = "1.0.0"

    override fun isNew() = true
    private val logMsg by lazy {
        (1..8).toList().joinToString("\n") { "$it. 修复第${it}个bug" } +
                "\n[点击查看详情](https://juejin.cn/user/817692382075102/posts)"
    }
    var msg: String? = null
    override fun getLog(): String {
        return msg ?: logMsg
    }
}